torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -80,7 +80,11 @@ def __init__(
         # However, this means that the shape of input bucketized tensors is subject to change,
         # which will complicate the implementations of error feedback and warm-up.
         # Running vanilla allreduce in the first few iterations can avoid this complexity.
-        assert start_powerSGD_iter >= 1
+        if use_error_feedback or warm_start:
```

Comment:
Have we checked all internal callsites to ensure that this won't result in a crash for users who are using the hook? 
Also, would raising an exception here rather than an assert be more appropriate? Either way, would also be useful to add a test that ensures an error is appropriately raised when this path is hit.

Reply:
> Have we checked all internal callsites to ensure that this won't result in a crash for users who are using the hook?

There is no internal callsite so far.

> Also, would raising an exception here rather than an assert be more appropriate? Either way, would also be useful to add a test that ensures an error is appropriately raised when this path is hit.

Replaced the assertion by exception, and added a unit test `test_invalid_powerSGD_state`.